portfolio/src/main/webapp/script.js

```diff
@@ -260,25 +260,66 @@ async function initMap() {
         addMarker(event.latLng);
     });
 
+    displayMarkers();
+
+    var input = document.getElementById("place-input");
+    var searchBox = new google.maps.places.SearchBox(input);
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(input);
+
+    // Restrict the search result near to the current viewport
+    map.addListener("bounds_changed", function() {
+        searchBox.setBounds(map.getBounds());
+    });
+
+    // Place a new marker on the place that the user searches
+    searchBox.addListener("places_changed", function() {
+        var places = searchBox.getPlaces();
+        if (places.length === 0) {
+            return;
+        }
+
+        var bounds = new google.maps.LatLngBounds();
+        places.forEach(function(place) {
+            if (!place.geometry) {
+                return;
+            }
+            if (place.geometry.viewport) {
+                bounds.union(place.geometry.viewport);
+            } else {
+                bounds.extend(place.geometry.location);
+            }
+        });
+        map.fitBounds(bounds);
+    });
+}
+
+async function displayMarkers() {
+    markers = [];
+    displayMarker(40.4382538, -79.9201019);
+    displayMarker(40.4376751, -79.9190924);
+
     const response = await fetch("/markers");
-    const markers = await response.json();
+    const responseMarkers = await response.json();
 
-    for (var index = 0; index < markers.length; index++) {
-        var latLng = markers[index];
+    for (var index = 0; index < responseMarkers.length; index++) {
+        var latLng = responseMarkers[index];
         var lat = parseFloat(latLng.latitude);
         var lng = parseFloat(latLng.longitude);
```

Comment:
If you want, you can use [destructuring assigment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) to unpack this object into two variables in one statement, instead of three.